tests/e2e/Services/Databases/DatabasesCustomClientTest.php

```diff
@@ -316,4 +316,338 @@ public function testUpdateTwoWayRelationship(): void
         $this->assertEquals($relation['body']['relatedCollection'], $collection1RelationAttribute['relatedCollection']);
         $this->assertEquals('restrict', $collection1RelationAttribute['onDelete']);
     }
+
+    public function testUpdateWithoutRelationPermission(): void
+    {
+        $userId = $this->getUser()['$id'];
+        $database = $this->client->call(Client::METHOD_POST, '/databases', [
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ], [
+            'databaseId' => ID::unique(),
+            'name' => ID::unique(),
+        ]);
+
+        $databaseId = $database['body']['$id'];
+
+        // Creating collection 1
+        $collection1 = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'collectionId' => ID::custom('collection1'),
+            'name' => ID::custom('collection1'),
+            'documentSecurity' => false,
+            'permissions' => [
+                Permission::create(Role::user($userId)),
+                Permission::read(Role::user($userId)),
+                Permission::delete(Role::user($userId)),
+            ]
+        ]);
+
+        // Creating collection 2
+        $collection2 = $this->client->call(Client::METHOD_POST, '/databases/' . $databaseId . '/collections', array_merge([
+            'content-type' => 'application/json',
+            'x-appwrite-project' => $this->getProject()['$id'],
+            'x-appwrite-key' => $this->getProject()['apiKey']
+        ]), [
+            'collectionId' => ID::custom('collection2'),
+            'name' => ID::custom('collection2'),
+            'documentSecurity' => false,
+            'permissions' => [
+                Permission::read(Role::user($userId)),
```

Comment:
Is there any test where a document in `collection1` is updated to link to a different `collection2` document? When I tried:

```json
{
    "data": {
        "level2": "level2b"
    }
}
```

I got an error:

```json
{
    "message": "The current user is not authorized to perform the requested action.",
    "code": 401,
    "type": "user_unauthorized",
    "version": "dev",
    "file": "/usr/src/code/app/controllers/api/databases.php",
    "line": 3333,
    "trace": [
        {
            "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
            "line": 593,
            "function": "{closure}",
            "args": [
                "one-to-one",
                "level1",
                "level1a",
                {
                    "$internalId": "1",
                    "$id": "level1a",
                    "$createdAt": "2023-08-15T00:50:55.532+00:00",
                    "$updatedAt": "2023-08-15T01:04:34.280+00:00",
                    "$permissions": [],
                    "$collection": "database_1_collection_1",
                    "level2": "level2b"
                },
                [],
                null,
                {},
                {},
                {},
                "default"
            ]
        },
        {
            "file": "/usr/src/code/vendor/utopia-php/framework/src/App.php",
            "line": 770,
            "function": "execute",
            "class": "Utopia\\App",
            "type": "->",
            "args": [
                {},
                {}
            ]
        },
        {
            "file": "/usr/src/code/app/http.php",
            "line": 248,
            "function": "run",
            "class": "Utopia\\App",
            "type": "->",
            "args": [
                {},
                {}
            ]
        }
    ]
}
```

Interestingly, when I updated like:

```json
{
    "data": {
        "level2": {
            "$id": "level2b"
        }
    }
}
```

it screwed up the document such that `$createdAt` and `$updatedAt"` disappeared and `_createdAt` and `_updatedAt` were added.

```json
{
    "$id": "level1a",
    "$createdAt": "2023-08-15T00:50:55.532+00:00",
    "$updatedAt": "2023-08-15T01:07:00.060+00:00",
    "$permissions": [],
    "level2": {
        "_createdAt": null,
        "_updatedAt": null,
        "text": null,
        "$id": "level2b",
        "$permissions": [],
        "level3": null,
        "$databaseId": "one-to-one",
        "$collectionId": "level2"
    },
    "$databaseId": "one-to-one",
    "$collectionId": "level1"
}
```

Reply:
> Do we have an issue or something to track this so that we can make a fix?

We don't have an issue open for this. Let me create an issue for that.

Reply:
https://github.com/appwrite/appwrite/issues/5999 
Added an issue